ocaml/xapi/xapi_vm_helpers.ml

```diff
@@ -933,10 +933,13 @@ let get_SRs_required_for_recovery ~__context ~self ~session_to =
 
 
 (* BIOS strings *)
-let assert_valid_bios_strings ~__context ~bios_strings =
+let assert_valid_bios_strings ~__context ~value =
+  (* Validate BIOS string keys *)
   (* Validate size of value provided is within bios_string_limit_size and not empty *)
   (* Validate value chars are printable ASCII characters *)
-  bios_strings |> List.iter (fun (k, v) ->
+  value |> List.iter (fun (k, v) ->
+    if not (List.mem k Xapi_globs.settable_vm_bios_string_keys) then
+      raise (Api_errors.Server_error(Api_errors.invalid_value, [k]));
```

Comment:
The `invalid_value` error is defined as `error Api_errors.invalid_value ["field"; "value"]`, so it needs two arguments.